oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Commit.java

```diff
@@ -354,7 +360,27 @@ private void applyToDocumentStore(RevisionVector baseBranchRevision)
         boolean commitRootHasChanges = operations.containsKey(commitRootPath);
         for (UpdateOp op : operations.values()) {
             NodeDocument.setCommitRoot(op, revision, commitRootDepth);
+
+            // special case for :childOrder updates
+            if (!previousRevisions.isEmpty()) {
+                boolean removePreviousSetOperations = false;
+                for (Map.Entry<Key, Operation> change : op.getChanges().entrySet()) {
+                    if (":childOrder".equals(change.getKey().getName()) && Operation.Type.SET_MAP_ENTRY == change.getValue().type) {
+                        // we are setting child order, so we should remove previous set operations from the same branch
+                        removePreviousSetOperations= true;
+                    }
+                }
+
+                if (removePreviousSetOperations) {
+                    for (Revision rev : previousRevisions) {
+                        op.removeMapEntry(":childOrder", rev);
+                    }
+                    LOG.debug("edited op is: " + op);
```

Comment:
I would log it at `INFO` level.

Reply:
I worry if this could result in too many messages if we log as `INFO` level, I think it's good as `DEBUG`.